internal/clients/github/client.go

```diff
@@ -360,6 +368,137 @@ func (c *client) GetActionsTokenPermissions(url string) (*types.TokenPermissions
 	return &p, nil
 }
 
+func (c *Client) IsAnalyzable(repository commontypes.RepositoryWithOwner) (bool, error) {
+	var repo struct {
+		Repository struct {
+			ViewerPermission githubv4.String
+		} `graphql:"repository(owner: $owner, name: $name)"`
+	}
+	variables := map[string]interface{}{
+		"name":  githubv4.String(repository.Name),
+		"owner": githubv4.String(repository.Owner),
+	}
+
+	err := c.GraphQLClient().Query(c.context, &repo, variables)
+	if err != nil {
+		return false, err
+	}
+
+	return repo.Repository.ViewerPermission == permissions.RepoRoleAdmin, nil
```

Comment:
does it work when user is OrgAdmin but not a member of the repo?

Reply:
yes